packages/xstate-fsm/src/types.ts

```diff
@@ -136,10 +136,14 @@ export namespace StateMachine {
     states: {
       [key in TState['value']]: {
         on?: {
-          [K in TEvent['type']]?: SingleOrArray<
+          [K in TEvent['type'] | '*']?: SingleOrArray<
             Transition<
               TContext,
-              TEvent extends { type: K } ? TEvent : never,
+              K extends '*'
```

Comment:
i think that by routing this earlier it becomes a little bit more readable, so I'd use this variant:
```ts
      on?: {
        [K in TEvent["type"] | "*"]?: SingleOrArray<
          K extends "*"
            ? Transition<TContext, TEvent, TState["value"]>
            : Transition<
                TContext,
                TEvent extends { type: K } ? TEvent : never,
                TState["value"]
              >
        >;
      };
```

Reply:
👍